src/guidescanpy/commands/add_sam_tag.py

```diff
@@ -0,0 +1,65 @@
+import argparse
+import pysam
+
+supported_tags = ["ce"]
+
+
+def get_parser(parser):
+    parser.add_argument("input", type=str, help="Path to the input sam/bam file.")
+
+    parser.add_argument(
+        "--ce",
+        action="store_true",
+        help="Add cutting-efficiency tag to the sam/bam file.",
+    )
+
+    parser.add_argument("--output", "-o", help="Path to the output sam file.")
+
+    parser.add_argument(
+        "--output-format",
+        choices=["bam", "sam"],
+        help="Specify the output file format: bam or sam",
+        default="bam",
+    )
+    return parser
+
+
+def add_tag(tag_name, input_sam_file, output_sam_file, is_bam):
+    if is_bam:
+        writing_mode = "wb"
+    else:
+        writing_mode = "w"
+
+    with pysam.AlignmentFile(input_sam_file, "r") as input_file, pysam.AlignmentFile(
```

Comment:
Make sure that "r" works for input BAM files and we don't need "rb"

Reply:
Somehow 'r' works for both `.bam` files and `.sam` files in my environment, but it's definetely better to use `rb` for `bam` to make it less confusing.

The official documentation says:
>If mode is not specified, the method will try to auto-detect in the order ‘rb’, ‘r’, thus both the following should work:
>```
>f1 = pysam.AlignmentFile('ex1.bam')
>f2 = pysam.AlignmentFile('ex1.sam')
>```

Reply:
Actually it's better to leave the reading mode blank. pysam's auto-detection is more accurate than our get_format function.